packages/toolpad-components/src/DataGrid.tsx

```diff
@@ -293,10 +317,55 @@ const DataGridComponent = React.forwardRef(function DataGridComponent(
     [selection?.id],
   );
 
-  const columns: GridColumns = React.useMemo(
-    () => (columnsProp ? parseColumns(columnsProp) : []),
-    [columnsProp],
-  );
+  const columns: GridColumns = React.useMemo(() => {
```

Comment:
not sure about using an effect to alter the dom from the component - is it not possible to set these columns as initial values according to whether there is an `onDelete` prop or not?

Reply:
but i guess this might be the best way, if the prop can be changed... just wondering if there's another way.